pubtools/pulplib/_impl/model/maintenance.py

```diff
@@ -0,0 +1,206 @@
+import logging
+import datetime
+import os
+
+import jsonschema
+
+from pubtools.pulplib._impl import compat_attr as attr
+from ..schema import load_schema
+from .common import InvalidDataException
+
+
+LOG = logging.getLogger("pubtools.pulplib")
+
+
+USER = os.environ.get("USER")
+HOSTNAME = os.environ.get("HOSTNAME")
+
+
+def iso_time_now():
+    return datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+@attr.s(kw_only=True, frozen=True)
+class MaintenanceEntry(object):
+    """Details about the maintenance status of a specific repository."""
+
+    repo_id = attr.ib(type=str)
+    """ID of repository in maintenance.
+
+    Note: there is no guarantee that a repository of this ID currently exists
+    in the Pulp server."""
+    message = attr.ib(default=None, type=str)
+    """Why this repository is in maintenance."""
+    owner = attr.ib(default=None, type=str)
+    """Who set this repository in maintenance mode."""
+    started = attr.ib(default=None, type=datetime.datetime)
+    """:class:`~datetime.datetime` in UTC at when the maintenance started."""
+
+
+@attr.s(kw_only=True, frozen=True)
+class MaintenanceReport(object):
+    """Represents the maintenance status of Pulp repositories.
+
+    On release-engineering Pulp servers, it's possible to put individual repositories
+    into "maintenance mode".  When in maintenance mode, external publishes of a repository
+    will be blocked.  Other operations remain possible.
+
+    This object holds information on the set of repositories currently in maintenance mode.
+    """
+
+    _OWNER = "%s@%s" % (USER, HOSTNAME) if all([USER, HOSTNAME]) else "pubtools.pulplib"
+
+    _SCHEMA = load_schema("maintenance")
+
+    last_updated = attr.ib(default=iso_time_now(), type=datetime.datetime)
```

Comment:
I see, having default=now will have the same problem here as I mentioned for MaintenanceEntry class.  Two identical statements would produce different objects depending on timing.  Can we please have a None default here too?

```suggestion
    last_updated = attr.ib(default=None, type=datetime.datetime)
```